micro-rdk/src/common/webrtc/api.rs

```diff
@@ -283,40 +287,51 @@ impl WebRtcSignalingChannel {
-            Some(req) => {
-                let req = req?;
-                if let Some(stage) = req.stage {
-                    match stage {
-                        answer_request::Stage::Update(c) => {
-                            if let Some(c) = c.candidate {
-                                log::debug!("received candidate {}", c.candidate);
-                                return c
-                                    .candidate
-                                    .try_into()
-                                    .map_err(|_| WebRtcError::CannotParseCandidate)
-                                    .map(Option::Some);
-                            } else {
-                                log::error!("received no candidates with this update request");
+        // Loop to allow receiving heartbeats without returning the next remote candidate.
+        loop {
+            match self.signaling_rx.next().await {
+                None => {
+                    return Err(WebRtcError::SignalingDisconnected());
+                }
+                Some(req) => {
+                    let req = req?;
+                    if let Some(stage) = req.stage {
+                        match stage {
+                            answer_request::Stage::Update(c) => {
+                                if let Some(c) = c.candidate {
+                                    log::debug!("received candidate {}", c.candidate);
+                                    return c
+                                        .candidate
+                                        .try_into()
+                                        .map_err(|_| WebRtcError::CannotParseCandidate)
+                                        .map(Option::Some);
+                                } else {
+                                    log::error!("received no candidates with this update request");
+                                    return Ok(None);
+                                }
+                            }
+                            answer_request::Stage::Error(s) => {
+                                if let Some(status) = s.status {
+                                    return Err(WebRtcError::SignalingError(status.message));
+                                }
+                                return Err(WebRtcError::SignalingError("unknown".to_owned()));
+                            }
+                            answer_request::Stage::Done(_) => {
                                 return Ok(None);
                             }
-                        }
-                        answer_request::Stage::Error(s) => {
-                            if let Some(status) = s.status {
-                                return Err(WebRtcError::SignalingError(status.message));
+                            answer_request::Stage::Heartbeat(_) => {
+                                log::debug!("received a heartbeat from the signaling server");
+                                continue;
```

Comment:
do we want this to potentially continue indefinitely? If not, I think we would want some max_iterations

Reply:
I don't know about continue _indefinitely_ but I don't _think_ we want a max iteration here. My understanding is that this `loop` will receive the next `AnswerRequest` for the current signaling exchange. If it's a heartbeat, it should try to call `next()` on the `signaling_rx` again. If the signaling server has been disconnected, we should be able to exit the loop via L294 (`return Err(WebRtcError::SignalingDisconnected());`). If the exchange times out, we should be able to exit the loop via L315, as the signaling server will send us an `answer_request::Stage::Error` if there was a timeout.

Reply:
This logic checks out to me, can resolve